# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../udftools/hot_fixes.patch
# Copyright (C) 2009 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

Submitted By: Guy Dalziel <gdalziel at linuxfromscratch dot org>
Date: 2009-08-06 
Initial Package Version: 1.0.0b3
Upstream Status: Sent
Origin: Debian, Dan Nicholson, and Guy Dalziel 
Description: Fixes compilation issues with gcc-4:
               wrudf.c:248: error: lvalue required as left operand of assignment
               cdrwtool.c:606: error: 'INT_MAX' undeclared (first use in this function)
               pktsetup.c:66: error: 'INT_MAX' undeclared (first use in this function)

diff -Naur ./cdrwtool/cdrwtool.c.orig ./cdrwtool/cdrwtool.c
--- ./cdrwtool/cdrwtool.c.orig	2002-12-28 04:48:51.000000000 +0000
+++ ./cdrwtool/cdrwtool.c	2009-08-06 12:41:08.909353809 +0100
@@ -27,6 +27,7 @@
 #include <unistd.h>
 #include <getopt.h>
 #include <signal.h>
+#include <limits.h>
 
 #include <sys/ioctl.h>
 #include <asm/param.h>
diff -Naur ./pktsetup/pktsetup.c.orig ./pktsetup/pktsetup.c
--- ./pktsetup/pktsetup.c.orig	2002-11-26 07:18:51.000000000 +0000
+++ ./pktsetup/pktsetup.c	2009-08-06 12:41:31.864951075 +0100
@@ -21,6 +21,7 @@
 #include <sys/ioctl.h>
 #include <unistd.h>
 #include <getopt.h>
+#include <limits.h>
 #include <bits/types.h>
 #include <sys/types.h>
 
diff -Naur ./wrudf/wrudf.c.orig ./wrudf/wrudf.c
--- ./wrudf/wrudf.c.orig	2002-11-26 07:18:51.000000000 +0000
+++ ./wrudf/wrudf.c	2009-08-06 12:40:50.258305159 +0100
@@ -245,7 +245,7 @@
 	    } else if( strncmp( spm->partIdent.ident, UDF_ID_VIRTUAL, strlen(UDF_ID_VIRTUAL)) == 0 )
 		virtualPartitionNum = i;
 	}
-	(char*)spm += spm->partitionMapLength;
+	spm = (char*)spm + spm->partitionMapLength;
     }
 
     if( medium == CDR ) {
